core/src/test/scala/caliban/parsing/ParserSpec.scala

```diff
@@ -492,6 +492,55 @@ object ParserSpec extends DefaultRunnableSpec {
           )
         )
       },
+      testM("input with no body") {
+        val inputWithNoBody = "input BarBaz".stripMargin
+        assertM(Parser.parseQuery(inputWithNoBody))(
+          equalTo(
+            Document(
+              List(
+                InputObjectTypeDefinition(
+                  description = None,
+                  name = "BarBaz",
+                  directives = Nil,
+                  fields = Nil
+                )
+              ),
+              sourceMapper = SourceMapper.apply(inputWithNoBody)
+            )
+          )
+        )
+      },
+      testM("input with empty body") {
+        val inputWithEmptyBody = "input BarBaz { }"
+        assertM(Parser.parseQuery(inputWithEmptyBody))(
+          equalTo(
+            Document(
+              List(
+                InputObjectTypeDefinition(
+                  description = None,
+                  name = "BarBaz",
+                  directives = Nil,
+                  fields = Nil
+                )
+              ),
+              sourceMapper = SourceMapper.apply(inputWithEmptyBody)
+            )
+          )
+        )
+      },
+      testM("enum with no body") {
+        val enumWithNoBody = "enum BarBaz"
```

Comment:
This test now fails, with the message

```
 - enum with no body
    Fiber failed.
    A checked error was not handled.
    Parsing Error: Position 1:12, found "" 
```

I think this is something to do with the grammar still being ambiguous between the input/enums with and without the set of braces at all.

The scalar type parser is largely the same as the parsers for the no-body input/enums, and it works just fine, which further leads me to believe that there's an issue going on with the nearly-identical tokens at the start of input/enum.

In any case, I can't work on this more tonight, it's getting late here in Vancouver 😅

Reply:
Maybe instead of 2 different parsers, just have one with the {...} part being optional?